src/Notes.cc

```diff
@@ -339,12 +339,12 @@ static void
 AppendTokens(NotePairs::Entries &entries, const SBuf &key, const SBuf &val, const CharacterSet &delimiters)
 {
     Parser::Tokenizer tok(val);
-    SBuf v;
-    while (tok.token(v, delimiters))
-        entries.push_back(new NotePairs::Entry(key, v));
-    v = tok.remaining();
-    if (!v.isEmpty())
-        entries.push_back(new NotePairs::Entry(key, v));
+    const auto tokenCharacters = delimiters.complement("non-delimiters");
```

Comment:
Computing this complement is a fairly expensive operation. Its result cannot be cached because `delimiters` may change from one call to another. If we want to optimize this and similar code, I see two options:

1. Add Tokenizer::prefixUntil() or similar method that takes delimiters instead of their complement.
2. Make CharacterSet::compliment() fast, either by always storing and maintaining complement (doubling memory usage requirement) or by storing a "reverse every comparison" flag (slowing down every comparison a little). In either case, we would have to make CharacterSet::Storage shareable across CharacterSets (to make CharacterSet copies cheap).

The first option looks significantly better to me because it does not slow down every CharacterSet test. In other words, the first option does not hurt any caller performance, while the second option hurts every caller performance (a little). The first option is also much easier to implement, and it does not make the rest of the affected class code more complex.

I consider this optimization, regardless of the implementation option, outside _this_ bug-fixing PR scope.